nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -189,7 +189,92 @@ in
       };
 
       buildMachines = mkOption {
-        type = types.listOf types.attrs;
+        type = types.loaOf (submodule ({ config }: {
+          options = {
+            sshUser = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              description = ''
+                The user as who to SSH to the build machine.
+              '';
+            };
+            hostName = mkOption {
+              type = types.string;
+              description = ''
+                The hostname of the build machine.
+              '';
+            };
+            system = mkOption {
+              type = types.string;
+              default = null;
+              description = ''
+                The system type the build machine can execute derivations on.
+                null if multiple are supported.
+              '';
+            };
+            systems = mkOption {
+              type = types.listOf types.string;
+              default = [];
+              description = ''
+                The system types the build machine can execute derivations on.
+              '';
+            };
+            sshKey = mkOption {
+              type = types.string;
+              default = "-";
+              description = ''
+                The path to the SSH private key with which to authenticate with
+                the build machine. <literal>"-"</literal> indicates falling back
+                on defaults.
+              '';
+            };
+            maxJobs = mkOption {
+              type = types.int;
+              default = 1;
+              description = ''
+                The number of concurrent jobs the build machine supports. The
+                build machine will enforce its own limits but this allows hydra
+                to schedule better since there is no work-stealing between build
+                machines.
+              '';
+            };
+            speedFactor = mkOption {
+              type = types.int;
+              default = 1;
+              description = ''
+                Something at indicates how fast the machine is relative to an
+                arbitrary norm???
+              '';
+            };
+            mandatoryFeatures = mkOptions {
+              type = types.listOf types.string;
+              default = [];
+              decriptions = ''
+                A list of features derivations built with this remote are
+                required to opt into using. (See the documentation on Nix itself
+                for what those features are.)
+              '';
+            };
+            supportedFeatures = mkOptions {
+              type = types.listOf types.string;
+              default = [];
+              decriptions = ''
+                A list of features derivations built with this remote may choose
+                to use or not. (See the documentation on Nix itself for what
+                those features are.)
```

Comment:
What do you mean by the above parentheticals?

Reply:
I was hoping somewhere else might actually list valid features, but https://nixos.org/nix/manual/#chap-distributed-builds again just says they exist, other than the "kvm" example.

Reply:
Ah, there is no list of valid features. You can make them up. I have a machine with the feature "big-mistake-this-ones-impure".

Reply:
oh haha

Reply:
a grep on nixpkgs gives "big-parallel"/"kvm"/"benchmark"/"nixos-test". Even if it stays freefrom, it would be nice to list the default ones with a types.either( enum [ "kvm" ... ]   or string).